pkg/controller/history/controller_history.go

```diff
@@ -244,20 +247,25 @@ func (rh *realHistory) CreateControllerRevision(parent metav1.Object, revision *
 	}
 	clone := any.(*apps.ControllerRevision)
 
+	// First attempt to create the revision will not use collisionCount
```

Comment:
Regarding item 1, there's no reason for that. It was meant to be consistent with the origin implementation that used a internal probe. I will change it to use it always.